docs/topics/class-methods.rst

```diff
@@ -0,0 +1,123 @@
+===========================
+Class Factory Methods
+===========================
+
+Factory methods create an instance of their implementer class by 
+extracting the components needed for it from the argument that the method takes.
+Throughout Scrapy the most common factory methods are ``from_crawler`` and ``from_settings`` where 
+they each take one parameter namely, a crawler or a settings object respectively.
+
+
+The ``from_crawler`` class method is implemented in the following objects:
+    * ItemPipeline
+    * DownloaderMiddleware
+    * SpiderMiddleware
+    * Scheduler
+    * BaseScheduler
+    * Spider
+
+The ``from_settings`` class method is implemented in the following objects:
+    * MailSender
+    * SpiderLoader
```

Comment:
Some of these are classes, some are interface definitions (they define an API, but do not really exist as classes in Python code), but none of them are objects.

This also seems to suggest that the upper list supports only `from_crawler` and the lower list supports only `from_settings`. Is that so?